src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -416,6 +419,37 @@ void InitCapabilities() {
 
 }  // namespace
 
+namespace {
+/**
+ * @brief Saves smart object content into the JSON node
+ * @param field_name name of the field to save
+ * @param schema reference to schema to unapply
+ * @param object_to_save pointer to object to save
+ * @param out_json_node JSON node for the output result
+ */
+void save_hmi_capability_field_to_json(
+    const std::string& field_name,
+    const smart_objects::CSmartSchema& schema,
+    smart_objects::SmartObjectSPtr object_to_save,
+    Json::Value& out_json_node) {
+  if (!object_to_save) {
+    return;
+  }
+
+  namespace Formatters = ns_smart_device_link::ns_json_handler::formatters;
+  smart_objects::SmartObject formatted_object(smart_objects::SmartType_Map);
+  formatted_object[strings::msg_params][field_name] = *object_to_save;
+  formatted_object.setSchema(schema);
+  formatted_object.getSchema().unapplySchema(
+      formatted_object);  // converts enums back to strings
```

Comment:
@LitvinenkoIra Not clear, why we firstly set schema into an object and at the next step we get the schema from the object? Because is these methods we do the next actions:
```
void SmartObject::setSchema(const CSmartSchema& schema) {
  m_schema = schema;
}

CSmartSchema SmartObject::getSchema() {
  return m_schema;
}
```
We don't do any checks, any comparisons or some similar actions.
I suggest to do the next:
```
schema.unapplySchema( formatted_object );
```